src/Compilers/CSharp/Portable/CSharpResources.resx

```diff
@@ -5363,6 +5363,9 @@ To remove the warning, you can use /reference instead (set the Embed Interop Typ
   <data name="ERR_ExplicitNullableAttribute" xml:space="preserve">
     <value>Explicit application of 'System.Runtime.CompilerServices.NullableAttribute' is not allowed.</value>
   </data>
+  <data name="ERR_NullableUnconstrainedTypeParameter" xml:space="preserve">
+    <value>A nullable type parameter must have a value type or reference type constraint.</value>
```

Comment:
Consider something more specific to guide users:
"... must be known to be a value or reference type. Consider adding a `class`, `struct` or type constraint."